make/autoconf/toolchain.m4

```diff
@@ -389,6 +389,12 @@ AC_DEFUN_ONCE([TOOLCHAIN_POST_DETECTION],
   # This is necessary since AC_PROG_CC defaults CFLAGS to "-g -O2"
   CFLAGS="$ORG_CFLAGS"
   CXXFLAGS="$ORG_CXXFLAGS"
+
+  # filter out some unwanted additions autoconf may add to CXX; we saw this on macOS with autoconf 2.72
+  UTIL_GET_NON_MATCHING_VALUES(cxx_filtered, $CXX, -std=c++11 -std=gnu++11)
+  if test "x$cxx_filtered" != x; then
```

Comment:
Why this test? If CXX is empty, then xcc_filtered will be empty too, right? And if CXX is exactly `-std=c++11`, then this test will render cxx_filter empty too, which will not change CXX -- which I believe is not what you want?

Reply:
Probably you are right; I think I 'borrowed'  the pattern with the additional` if test` from some other places where UTIL_GET_NON_MATCHING_VALUES is used (e.g. jvm-features.m4).
Should I remove the  test ?

Reply:
I agree with @magicus. It should work when you then unconditionally assign cxx_filtered to CXX after the call.